### ISSUE TITLE
Medical armor exploit

### ISSUE THREAD

**Issue by epyon** [chaosforgeorg/doomrl#97]
Medical Armor heals 1% at the cost of 1 durability. With Nanofiber Skin Armor assembly, this allows the armor to self-repair after 5 actions of not taking damage. This delay can be reduced to 0 with a Nano Pack. With Nanofiber Skin Medical Armor (N), the player can essentially take 1 damage, heal 1 damage, the armor repairs 1 damage, and this can be done infinitely. The damage taken value could have adverse effects if it reaches the maximum value possible, such as rolling over to 0, which could be exploited to complete "damageless" challenges. This doesn't work with Onyx Packs, which simply hide durability, although it's still there. This should be tested first if any negative effects can actually occur, or otherwise change how the assembly works with Medical Armor, such as making the recharge delay higher and impossible to reach 0.

**Comment by OmegaTyrant** [comment]
Yeah I removed the proposal to "fix" this from the compilation list because I don't believe it to be a real problem. Even if you don't nano mod on top of the Nanofiber Skin Medical Armor, you could take damage until the armor enters it damage state healing you, get off the acid, wait until it repairs and heals you up, then get back on the acid and repeat to take as much damage as you want. You could also do this with just normal N-modded Medical Armor, or with Vampyre if you got a respawn or infinitely spawning enemies. So the "fix" to this would just be a pointless nerf to Medical Armor in this specific assembly. Damage overflow, if it happens, would also take an absurdly long time to achieve, and if it is a concern that someone can cheese damageless badges with a damage overflow, some other fix directly addressing the damage overflow should be implemented instead.

**Comment by OmegaTyrant** [comment]
We got [confirmation](https://forum.chaosforge.org/index.php/topic,9261.msg71584/topicseen.html#msg71584) that the damage counter doesn't overflow at 65535 and will just continue to keep counting, so I think we can close this potential "exploit" as a non-issue.

**Comment by brisbang** [comment]
Hey - I like that farm!
